api/deals/models.py

```diff
@@ -9,87 +22,46 @@ class DealModel(BaseModel):
     """
 
     order_id: int = Field(default=0)
-    buy_price: float = Field(default=0)
-    buy_total_qty: float = Field(default=0)
-    buy_timestamp: float = Field(default=0)
-    current_price: float = Field(default=0)
-    sd: float = Field(default=0)
-    avg_buy_price: float = Field(default=0)
-    take_profit_price: float = Field(default=0)
-    sell_timestamp: float = Field(default=0)
-    sell_price: float = Field(default=0)
-    sell_qty: float = Field(default=0)
-    trailling_stop_loss_price: float = Field(
+    current_price: amount = Field(default=0)
+    take_profit_price: amount = Field(
         default=0,
-        description="take_profit but for trailling, to avoid confusion, trailling_profit_price always be > trailling_stop_loss_price",
+        description="derived from take_profit, while this price gets updated according to market, take_profit percentage doesn't",
     )
-    trailling_profit_price: float = Field(default=0)
-    stop_loss_price: float = Field(default=0)
-    trailling_profit: float = Field(default=0)
-    original_buy_price: float = Field(
-        default=0, description="historical buy_price after so trigger"
+    trailling_stop_loss_price: amount = Field(
+        default=0,
+        description="take_profit but for trailling, to avoid confusion, trailling_profit_price always be > trailling_stop_loss_price",
     )
-    short_sell_price: float = Field(default=0)
+    trailling_profit_price: amount = Field(default=0)
+    stop_loss_price: amount = Field(default=0)
+    trailling_profit: amount = Field(default=0)
+    original_buy_price: amount = Field(default=0)
 
     # fields for margin trading
-    margin_short_loan_principal: float = Field(default=0)
+    margin_short_loan_principal: amount = Field(default=0)
     margin_loan_id: float = Field(default=0)
-    hourly_interest_rate: float = Field(default=0)
-    margin_short_sell_price: float = Field(default=0)
-    margin_short_loan_interest: float = Field(default=0)
-    margin_short_buy_back_price: float = Field(default=0)
-    margin_short_sell_qty: float = Field(default=0)
-    margin_short_buy_back_timestamp: int = 0
-    margin_short_base_order: float = Field(default=0)
-    margin_short_sell_timestamp: int = Field(default=0)
-    margin_short_loan_timestamp: int = Field(default=0)
+    hourly_interest_rate: amount = Field(
+        default=0,
+        description="to be deprecated, only used for interest calculation",
+    )
+    margin_short_loan_interest: amount = Field(default=0)
+    margin_short_loan_timestamp: amount = Field(default=0)
```

Comment:
Remove all loan fields. Storing `margin_loan_id` should be enough to get loan data from API.